src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/ProjectFileSynchronizerTest.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer
+{
+    public class ProjectFileSynchronizerTest : LanguageServerTestBase
+    {
+        [Fact]
+        public void ProjectFileChanged_Added_AddsProject()
+        {
+            // Arrange
+            var projectPath = "/path/to/project.csproj";
+            var projectService = new Mock<RazorProjectService>(MockBehavior.Strict);
+            projectService.Setup(service => service.AddProject(projectPath)).Verifiable();
+            var synchronizer = new ProjectFileSynchronizer(Dispatcher, projectService.Object);
+
+            // Act
+            synchronizer.ProjectFileChanged(projectPath, RazorFileChangeKind.Added);
+
+            // Assert
+            projectService.VerifyAll();
+        }
+
+        [Fact]
+        public void ProjectFileChanged_Removed_RemovesProject()
+        {
+            // Arrange
+            var projectPath = "/path/to/project.csproj";
+            var projectService = new Mock<RazorProjectService>(MockBehavior.Strict);
+            projectService.Setup(service => service.RemoveProject(projectPath)).Verifiable();
```

Comment:
Does `Verifiable` by default demand that anything that got setup gets called exactly once?

Reply:
Not exactly once but "at least once" Also as an extra tid bit `MockBehavior.Strict` above ensures that no method that wasn't setup gets called.

Reply:
I knew about Strict, but didn't know that Setup would force at least once by default. Neat!